src/openzaak/components/catalogi/admin/mixins.py

```diff
@@ -410,3 +417,25 @@ def get_concept(self, obj):
         if not obj:
             return True
         return obj.zaaktype.concept
+
+
+class NotificationMixin:
+    def save_model(self, request, obj, form, change):
+        if isinstance(obj, ZaakType):
+            viewset = ZaakTypeViewSet()
+        elif isinstance(obj, InformatieObjectType):
+            viewset = InformatieObjectTypeViewSet()
+        elif isinstance(obj, BesluitType):
+            viewset = BesluitTypeViewSet()
+
+        scheme = api_settings.DEFAULT_VERSIONING_CLASS()
+        request.version, request.versioning_scheme = (
+            scheme.determine_version(request, version=api_settings.DEFAULT_VERSION),
+            scheme,
+        )
```

Comment:
I'd prefer an approach similar to https://github.com/encode/django-rest-framework/blob/master/rest_framework/views.py#L385 where the `request` is wrapped in `rest_framework.request.Request`, rather than mutating the original `request` object.

Side-effects are not cool, try to avoid them!

Reply:
additionally, looking at https://github.com/encode/django-rest-framework/blob/master/rest_framework/views.py#L373 you should be able to rely on the viewset for the versioning scheme stuff:

```py
version, scheme = viewset.determine_version(request)
```